python/hail/expr/functions.py

```diff
@@ -1896,7 +1896,20 @@ def is_transversion(ref, alt) -> BooleanExpression:
     -------
     :class:`.BooleanExpression`
     """
-    return _func("is_transversion", tbool, ref, alt)
+    return is_snp(ref, alt) & (~(_is_snp_transition(ref, alt)))
+
+
+@typecheck(ref=expr_str, alt=expr_str)
+def _is_snp_transition(ref, alt) -> BooleanExpression:
+    i = 0
+    while i < hl.eval_expr(ref.length()):
```

Comment:
the loops and conditionals have to also be a part of the constructed expression.